modules/host/obligations.go

```diff
@@ -0,0 +1,264 @@
+	// data under the obligations is being protected by the host's mutex.
+	// Grabbing 'mu' is not sufficient to guarantee modification safety of the
+	// struct, the host mutex must also be grabbed.
+	mu sync.Mutex
+}
+
+// fileSize returns the size of the file that is held by the contract
+// obligation.
+func (co *contractObligation) fileSize() uint64 {
+	if co.hasRevision() {
+		return co.RevisionTxn.FileContractRevisions[0].NewFileSize
+	}
+	return co.OriginTxn.FileContracts[0].FileSize
+}
+
+// hasRevision indiciates whether there is a file contract reivision contained
+// within the contract obligation.
+func (co *contractObligation) hasRevision() bool {
+	return len(co.RevisionTxn.FileContractRevisions) == 1
+}
+
+// missedProofUnlockHash returns the operating unlock hash for a successful
+// file contract in the obligation.
+func (co *contractObligation) missedProofUnlockHash() types.UnlockHash {
+	if co.hasRevision() {
+		return co.RevisionTxn.FileContractRevisions[0].NewMissedProofOutputs[1].UnlockHash
+	}
+	return co.OriginTxn.FileContracts[0].MissedProofOutputs[1].UnlockHash
+}
+
+// payout returns the operating payout of the contract obligation.
+func (co *contractObligation) payout() types.Currency {
+	// Function seems unnecessary because it is just a getter, but adding this
+	// function helps maintain consistency with the way that the other fields
+	// are accessed.
+	return co.OriginTxn.FileContracts[0].Payout
+}
+
+// proofConfirmed inidicates whether the storage proofs have been seen on the
+// blockchain.
+func (co *contractObligation) proofConfirmed() bool {
+	// Function seems unnecessary because it is just a getter, but adding this
+	// function helps maintain consistency with the way that the other fields
+	// are accessed.
+	return co.ProofConfirmed
+}
+
+// reset updates the contract obligation to reflect that the consensus set is
+// being rescanned, which means all of the consensus indicators need to be
+// reset, and the action items need to be filled out again.
+func (co *contractObligation) reset() {
+	co.OriginConfirmed = false
+	co.ProofConfirmed = false
+
+	// If there is no revision, then the final revision counts as being
+	// confirmed. Otherwise, the revision should not be considered as
+	// confirmed.
+	if co.hasRevision() {
+		co.RevisionConfirmed = false
+	} else {
+		co.RevisionConfirmed = true
+	}
+}
+
+// revisionNumber returns the operating revision number of the obligation.
+func (co *contractObligation) revisionNumber() uint64 {
+	if co.hasRevision() {
+		return co.RevisionTxn.FileContractRevisions[0].NewRevisionNumber
+	}
+	return co.OriginTxn.FileContracts[0].RevisionNumber
+}
+
+// txnsConfirmed indiactes whether the file contract and its latest revision
+// have been seen on the blockchain.
+func (co *contractObligation) txnsConfirmed() bool {
+	if !co.OriginConfirmed {
+		return false
+	}
+	if co.hasRevision() && !co.RevisionConfirmed {
+		return false
+	}
+	return true
```

Comment:
I find my way to be more intuitive. I've added more comments.

If the first txn isn't confirmed, then the txns aren't confirmed
if the second txn exists but isn't confirmed, then the txns aren't confirmed
otherwise, the txns are confirmed.